src/Interpreters/MutationsInterpreter.cpp

```diff
@@ -402,10 +402,17 @@ MutationsInterpreter::MutationsInterpreter(
     , metadata_snapshot(metadata_snapshot_)
     , commands(std::move(commands_))
     , available_columns(std::move(available_columns_))
-    , context(Context::createCopy(context_))
     , settings(std::move(settings_))
     , select_limits(SelectQueryOptions().analyze(!settings.can_execute).ignoreLimits())
 {
+    auto new_context = Context::createCopy(context_);
+    if (new_context->getSettingsRef().allow_experimental_analyzer)
+    {
+        new_context->setSetting("allow_experimental_analyzer", false);
+        LOG_DEBUG(&Poco::Logger::get("MutationsInterpreter"), "Will use old analyzer to prepare mutation");
+    }
+    context = std::move(new_context);
```

Comment:
ok, but maybe we can disable analyzer only here?
https://github.com/ClickHouse/ClickHouse/blob/836e1963c4525757bc5580b0fecf9fbbed1fcfb0/src/Interpreters/MutationsInterpreter.cpp#L1077-L1082

e.g. change the context we pass to TreeRewriter/ExpressionAnalyzer

Reply:
I think it's just safer to disable it completely.